cli/lib/nf3.mjs

```diff
@@ -76,7 +77,12 @@ class Nf3 {
   */
   async init() {
     this.setWeb3Provider(this.web3WsUrl);
-    this.zkpKeys = this.zkpKeys || (await axios.post(`${this.clientBaseUrl}/generate-keys`)).data;
+    // Generate a random mnemonic (uses crypto.randomBytes under the hood), defaults to 128-bits of entropy
+    const mnemonic = generateMnemonic();
+    this.zkpKeys =
+      this.zkpKeys ||
+      (await axios.post(`${this.clientBaseUrl}/generate-keys`),
```

Comment:
The last ‘0’ stands for address index as explained in the PR. For the same mnemonic, we need to give different paths to generate different keys. So we increment the address index each time to generate the new keys. Because this is the first set of keys we started at 0

Reply:
I don't think i explained clearly what I meant. I believe your code is incorrect:
```
this.zkpKeys =
      this.zkpKeys ||
      (await axios.post(`${this.clientBaseUrl}/generate-keys`),
      { mnemonic, path: `m/44'/60'/0'/0` }).data;
```
If you wirte it like this, the mnemonic and path will not be passed to the end point (at least this happened on my tests). You need to include the mnemonic and path inside the post as:
```
this.zkpKeys =
      this.zkpKeys ||
      (await axios.post(`${this.clientBaseUrl}/generate-keys`,
      { mnemonic, path: `m/44'/60'/0'/0` })).data;
```

Reply:
Good spot. Fixed. Thank you